api/src/main/java/io/strimzi/api/kafka/model/CruiseControlSpec.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.fabric8.kubernetes.api.model.ResourceRequirements;
+import io.strimzi.api.kafka.model.balancing.BrokerCapacity;
+import io.strimzi.api.kafka.model.template.CruiseControlTemplate;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Maximum;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "replicas", "image", "config",
+        "livenessProbe", "readinessProbe",
+        "jvmOptions", "resources",
+        "logging", "tlsSidecar", "template", "brokerCapacity"})
+@EqualsAndHashCode
+public class CruiseControlSpec implements UnknownPropertyPreserving, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    // For the full configuration list refer to https://github.com/linkedin/cruise-control/wiki/Configurations
+    public static final String FORBIDDEN_PREFIXES = "bootstrap.servers, client.id, zookeeper., network., security., failed.brokers.zk.path,"
+        + "webserver.http., webserver.api.urlprefix, webserver.session.path, webserver.accesslog., two.step., request.reason.required,"
+        + "metric.reporter.sampler.bootstrap.servers, metric.reporter.topic, partition.metric.sample.store.topic, broker.metric.sample.store.topic,"
+        + "capacity.config.file, self.healing.";
+    public static final String FORBIDDEN_PREFIX_EXCEPTIONS = "";
+
+    private int replicas;
+    private String image;
+    private TlsSidecar tlsSidecar;
+    private ResourceRequirements resources;
+    private Probe livenessProbe;
+    private Probe readinessProbe;
+    private JvmOptions jvmOptions;
+    private Logging logging;
+    private CruiseControlTemplate template;
+    private BrokerCapacity brokerCapacity;
+    private Map<String, Object> config = new HashMap<>(0);
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("The number of pods in the `Deployment`.")
+    @Minimum(0)
+    @Maximum(1)
+    @JsonProperty(required = true)
+    public int getReplicas() {
+        return replicas;
```

Comment:
This allows scaling the CC pod to 0 to pause the CC application without removing the CC metric reporters from the Kafka brokers. This will stop CC from processing the raw metrics and sending them to Kafka. However, the CC metric reporters will continue to push raw metrics to Kafka when the CC application is paused.

Reply:
Does that sound reasonable? If not, we can revert it back to it being hardcoded

Reply:
Moved the `replicas` property out for now as it could potential confuse users into thinking that Cruise Control deployments can scale past 1 pod, it is inconsistent with the other component deployment configurations, and it adds another property that users would need to remember to deploy Cruise Control.

Now users can deploy Cruise Control with `cruiseControl: {}`

We can add it back in later if people find this functionality useful and want it!